### DISCUSSION TITLE
Why not use arm64v8/ubuntu when running on apple silicon?

### DISCUSSION THREAD

**Discussion by ShaneMurphy2** [cross-rs/cross#1360] [Q&A] [answered]
I'm pretty unfamiliar with docker, but is seems to me like the situation on M series macs involves running the build inside a QEMU instance shipped with binfmt. But, there exists https://hub.docker.com/r/arm64v8/ubuntu/ which would allow running natively, no? Curious as to if there is a blocker technically for this, as it seems like running in QEMU causes some headaches.

**Comment by Emilgardis** [comment, marked as answer]
See https://github.com/cross-rs/cross/issues/751, this is very high up on my wishlist.

Infact, the entire codebase ([aside from ci](https://github.com/cross-rs/cross/pull/849)) now supports this possibility, and you can preview how it would work with https://github.com/cross-rs/cross/issues/751#issuecomment-1432015974. [Here's](https://github.com/cross-rs/cross/issues/751#issuecomment-1776759292) also a more complete example without many of the bells our images have.

Many of our images actually work fine on [amd64 and arm64](https://github.com/cross-rs/cross/issues/975), the main issue is that github doesn't have native arm64 runners with docker so we'd have to use emulation to create the images.

also, we don't need that specific ubuntu image, `ubuntu` [is already multi-arch](https://hub.docker.com/layers/library/ubuntu/20.04/images/sha256-a80d11b67ef30474bcccab048020ee25aee659c4caaca70794867deba5d392b6?context=explore)

**Reply by ShaneMurphy2** [reply]
Thanks for the detailed reply. As expected, it's difficult. Goes without saying that the cross effort is a wonderful thing, and I don't under-estimate the complexity.
Excited to see this work!
